x/stakeibc/keeper/reward_converter.go

```diff
@@ -53,130 +59,73 @@ type ForwardMetadata struct {
-// The reward amount is denominated in the host zone's native denom - this is either directly from
-// staking rewards, or, in the case of a host zone with a trade route, this is called with the converted tokens
-// For instance, with dYdX, this is applied to the DYDX tokens that were converted from USDC
-//
-// If the chain doesn't have a rebate in place, the split is decided entirely from the stride commission percent
-// However, if the chain does have a rebate, we need to factor that into the calculation, by scaling
-// up the rewards to find the amount before the rebate
-//
-// For instance, if 1000 rewards were collected and 2 were sent as a rebate, then the stride fee should be based
-// on the original 1000 rewards instead of the remaining 998 in the query response:
-//
-//	Community pool liquid staked 1M, TVL is 10M, rebate is 20%, stride fee is 10%
-//	If 998 native tokens were queried, we have to scale that up to 1000 original reward tokens
-//
-//	Effective Rebate Pct = 10% fees * (1M LS / 10M TVL) * 20% rebate = 0.20% (aka 0.002)
-//	Effective Stride Fee Pct = 10% fees - 0.20% effective rebate = 9.8%
-//	Original Reward Amount = 998 Queried Rewards / (1 - 0.002 effective rebate rate) = 1000 original rewards
-//	Then stride fees are 9.8% of that 1000 original rewards = 98
-func (k Keeper) CalculateRewardsSplitAfterRebate(
-	ctx sdk.Context,
-	hostZone types.HostZone,
-	rewardsAmount sdkmath.Int,
-) (strideFeeAmount sdkmath.Int, reinvestAmount sdkmath.Int, err error) {
-	// Get the fee rate and total fees from params (e.g. 0.1 for 10% fee)
-	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
-	totalFeeRate := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
-
-	// Check if the chain has a rebate
-	rebateInfo, chainHasRebate := hostZone.SafelyGetCommunityPoolRebate()
+	rebateAmount := sdk.NewDecFromInt(totalFeesAmount).Mul(contributionRate).Mul(rebateInfo.RebateRate).TruncateInt()
+	strideFeeAmount := totalFeesAmount.Sub(rebateAmount)
 
-	// If there's no rebate, the fee split just uses the commission
-	if !chainHasRebate {
-		strideFeeAmount = sdk.NewDecFromInt(rewardsAmount).Mul(totalFeeRate).TruncateInt()
-	} else {
-		// Get supply of stTokens to determine the portion of TVL that the community pool liquid stake makes up
-		stDenom := utils.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
-		stTokenSupply := k.bankKeeper.GetSupply(ctx, stDenom).Amount
-
-		// It shouldn't be possible to have 0 token supply (since there are rewards and there was a community pool stake)
-		// This will also prevent a division by 0 error
-		if stTokenSupply.IsZero() {
-			return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrDivisionByZero,
-				"unable to calculate rebate amount for %s since total delegations are 0", hostZone.ChainId)
-		}
-
-		// It also shouldn't be possible for the liquid stake amount to be greater than the full TVL
-		if rebateInfo.LiquidStakedStTokenAmount.GT(stTokenSupply) {
```

Comment:
Should we keep these sanity checks? And some of the others e.g. `stTokenSupply.IsZero()`

Reply:
they are [still in there](https://github.com/Stride-Labs/stride/pull/1162/files/3ac87f999fb270a059ad67c22df73a23b4f529c1#diff-8850ef830f8cbe0e6ca0be787be1036bc793215c526fb43491a812ca4f0bb079R105-R114). 

They were previously in both the BeforeRebate and AfterRebate functions. But now there's only one function.